docs/import-from-whitehall.md

```diff
@@ -0,0 +1,110 @@
+
+Note that the ID in Whitehall's internal URLs is the edition number (e.g. for `/government/admin/publications/1034451` the edition is `1034451`).  The document ID can be obtained on a Rails console, using:
+
+```
+Edition.find(1034451).document.id
+```
+
+### Import documents of a specific type (and optional subtypes) from a single publishing organisation
+
+To import a single document type (e.g. news articles) from a single publisher (e.g. NDA):
+
+```
+rake import:whitehall_migration["nuclear-decommissioning-authority","news_article"]
+```
+
+To import multiple document subtypes (e.g. news stories and press releases) from a single publisher (e.g. NDA):
+
+```
+rake import:whitehall_migration["nuclear-decommissioning-authority","news_article","news_story,press_release"]
+```
+
+There is no limit to the number of subtypes (additional values are added comma separated).
+
+It is also possible to import a single document subtype (e.g. press releases) from a single publisher (e.g. NDA):
+
+```
+rake import:whitehall_migration["nuclear-decommissioning-authority","news_article","press_release"]
+```
+
+After running the rake task, a URL will be printed to the console which allows the status of import to be examined.
+
+## Running a migration locally using govuk-docker
+
+Using [govuk-docker](https://github.com/alphagov/govuk-docker), you can run an import locally without needing to replicate data.  In order to do this, you will need to override the [Plek](https://github.com/alphagov/plek) URL for Whitehall and Publishing API to point to staging.  Additionally, you will require the bearer tokens for these applications from [govuk-secrets](https://github.com/alphagov/govuk-secrets).
+
+### Getting bearer tokens
+
+Tokens can be obtained from govuk-secrets.  These allow access to the relevant remote applications without needing to enter a username and password.
+
+```
+cd govuk-secrets/puppet_aws
+bundle exec rake 'eyaml:edit[staging,apps]'
+```
+
+The values you need to obtain are:
+- `govuk::apps::content_publisher::whitehall_bearer_token` - the Whitehall bearer token
+- `govuk::apps::content_publisher::publishing_api_bearer_token` - the Publishing API bearer token
+
+### Setting up Sidekiq
+
+In order to run a migration locally, Sidekiq must be running when the rake tasks are executed.  As jobs are run independently of the rake task, Plek overrides and bearer tokens are also required here.
+
+```
+govuk-docker run -e PLEK_SERVICE_PUBLISHING_API_URI=https://publishing-api.staging.publishing.service.gov.uk -e PUBLISHING_API_BEARER_TOKEN=<> -e PLEK_SERVICE_WHITEHALL_ADMIN_URI=https://whitehall-admin.staging.publishing.service.gov.uk -e WHITEHALL_BEARER_TOKEN=<> content-publisher-worker bundle exec sidekiq -C config/sidekiq.yml
+```
+
+Replace `<>` with the bearer tokens you have previously obtained.
```

Comment:
I wonder if it'd be a bit clearer if we used example vars like $whitehall_bearer_token and $publishing_api_bearer_token.

Could also tell users how to export them first with 

```
export whitehall_bearer_token=<your token>
export publishing_api_bearer_token=<your token>
```

and then the longer commands could be pasted verbatim

Reply:
Updated throughout.  Also setting the URLs in the same way to reduce the length of the commands.